engine/framework/worker.go

```diff
@@ -82,20 +81,32 @@ type WorkerImpl interface {
 
 // BaseWorker defines the worker interface, it embeds a Worker interface and adds
 // more utility methods
+// TODO: decouple the BaseWorker and WorkerService(for business)
 type BaseWorker interface {
 	Worker
 
+	// MetaKVClient return business metastore kv client with job-level isolation
 	MetaKVClient() metaModel.KVClient
+
+	// MetricFactory return a promethus factory with some underlying labels(e.g. job-id, work-id)
 	MetricFactory() promutil.Factory
+
+	// Logger return a zap logger with some underlying fields(e.g. job-id)
 	Logger() *zap.Logger
+
+	// UpdateStatus persists the status to framework metastore if worker status is changed and
+	// sends 'status updated message' to master.
 	UpdateStatus(ctx context.Context, status frameModel.WorkerStatus) error
+
+	// SendMessage sends a message of specific topic to master in a blocking or nonblocking way
 	SendMessage(ctx context.Context, topic p2p.Topic, message interface{}, nonblocking bool) error
+
+	// OpenStorage creates a resource and return the resource handle
 	OpenStorage(ctx context.Context, resourcePath resourcemeta.ResourceID) (broker.Handle, error)
+
 	// Exit should be called when worker (in user logic) wants to exit.
-	// When `err` is not nil, the status code is assigned WorkerStatusError.
-	// Otherwise worker should set its status code to a meaningful value.
-	Exit(ctx context.Context, status frameModel.WorkerStatus, err error) error
-	NotifyExit(ctx context.Context, errIn error) error
+	// exitReason: ExitReasonFinished/ExitReasonCancelled/ExitReasonFailed
+	Exit(ctx context.Context, exitReason ExitReason, err error, errMsg string, extBytes []byte) error
```

Comment:
Can define a struct containing all exit information
```go
type ExitInfo struct {
    reason  ExitReason
    err error
    errMsg string
    extBytes []byte
}
```

Reply:
The reason why I don't do that is: 
 I want a unify params format for BaseMaster/BaseWorker.Exit()

Reply:
If using one single `ExitInfo`, the signature of `Exit` will be 
```go
func Exit(ctx context.Context, info ExitInfo) error
```
Is this more clean and easier for us to add functions to `ExitInfo`

Reply:
Now the function signature is: 
`Exit(ctx context.Context, exitReason ExitReason, err error, extBytes []byte) error`

Seems simple enough.